src/utils/collection.rs

```diff
@@ -0,0 +1,236 @@
+                "Could not get file name from path {}",
+                path.as_ref().display()
+            ),
+        };
+
+        Ok(BtfCollection {
+            base: NamedBtf {
+                name,
+                btf: Btf::from_file(path)?,
+            },
+            split: Vec::new(),
+        })
+    }
+
+    /// Construct a BtfCollection object from a base BTF file only.
+    pub fn from_bytes(name: &str, bytes: &[u8]) -> Result<BtfCollection> {
+        Ok(BtfCollection {
+            base: NamedBtf {
+                name: name.to_string(),
+                btf: Btf::from_bytes(bytes)?,
+            },
+            split: Vec::new(),
+        })
+    }
+
+    /// Add a split BTF in the current BtfCollection representation, reading a file.
+    pub fn add_split_btf_from_file<P: AsRef<Path>>(&mut self, path: P) -> Result<&mut Self> {
+        let name = match path.as_ref().file_name() {
+            Some(name) => match name.to_str() {
+                Some(s) => s.to_string(),
+                None => bail!("Invalid file name {:?}", name),
+            },
+            None => bail!(
+                "Could not get file name from path {}",
+                path.as_ref().display()
+            ),
+        };
+
+        if self.split.iter().any(|m| m.name == name) {
+            bail!("Split BTF with name {name} already present");
+        }
+
+        self.split.push(NamedBtf {
+            name,
+            btf: Btf::from_split_file(path, &self.base.btf)?,
+        });
+        Ok(self)
+    }
+
+    /// Add a split BTF in the current BtfCollection representation, reading a byte slice.
+    pub fn add_split_btf_from_bytes(&mut self, name: &str, bytes: &[u8]) -> Result<&mut Self> {
+        let name = name.to_string();
+        if self.split.iter().any(|m| m.name == name) {
+            bail!("Split BTF with name {name} already present");
+        }
+
+        self.split.push(NamedBtf {
+            name,
+            btf: Btf::from_split_bytes(bytes, &self.base.btf)?,
+        });
+        Ok(self)
+    }
+
+    /// Parse BTF objects stored in a directory and construct a BtfCollection
+    /// object, given a path to the directory and the filename of the base BTF file.
+    /// This is helpful for parsing /sys/kernel/btf for example.
+    pub fn from_dir<P: AsRef<Path>>(dir: P, base: &str) -> Result<BtfCollection> {
+        // First parse the base BTF information.
+        let mut sys_btf = BtfCollection::from_file(format!("{}/{base}", dir.as_ref().display()))?;
+
+        // Then loop over all split BTF files and parse them.
+        for file in fs::read_dir(dir.as_ref())? {
+            match file {
+                Ok(file) => {
+                    if file.file_name() == base {
+                        continue;
+                    }
+                    if let Ok(ft) = file.file_type() {
+                        if !ft.is_dir() {
+                            sys_btf.add_split_btf_from_file(file.path())?;
```

Comment:
Would it make sense to use the "from_bytes" flavor given we already have the file name?